src/column.rs

```diff
@@ -314,6 +339,8 @@ impl Column {
 					PlanOutcome::NeedReindex => {
 						log::debug!(target: "parity-db", "{}: Index chunk full {}", tables.index.id, hex(key));
 						Self::trigger_reindex(tables, reindex, self.path.as_path());
+						// [Review] Here we will call write_insert_plan a second time which seems fine, but
+						// could be skipped by just calling again tables.index.write_insert_plan?
```

Comment:
I believe that it is theoretically possible that `write_insert_plan` might return `PlanOutcome::NeedsReindex` the second time. Reindexing begins in the background so the chunk may be full again by the second call.

Reply:
Oh, my understanding on this point was that the function that trigger reindexing did switch the indexing to a new file with the new index bits.
Then write index is replayed and target the new index file.
While in background the reindex process just consume old data and write it to new file (which could trigger a reindex because of concurrent writes).
So even if all reindexing run before this log enactment, the size will only be half.

Reply:
> So even if all reindexing run before this log enactment, the size will only be half.

Unless after the reindex, all the items from the chunk still end up in a single chunk, as they all share the same additional bit. Which, I agree, is extremely unlikely.

Reply:
Oh yes, I did not think that way. So it is 'just' if the 64 entry got the same bit at the n + 1 positon, unlikely, but better checked :+1: